### ISSUE TITLE
Averaging is extremely slow in some setups

### ISSUE THREAD

**Issue by yhn112** [learning-at-home/hivemind#351]
Error log from client-mode peer:
```
[2021/08/16 22:54:50.049][INFO][optim.collaborative.step:229] Beginning global optimizer step 0
[2021/08/16 22:54:50.253][INFO][optim.collaborative.fetch_collaboration_state:444] Collaboration accumulated 3696 samples from 2 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/16 22:54:50.445][INFO][optim.collaborative.fetch_collaboration_state:444] Collaboration accumulated 3696 samples from 2 peers; ETA 0.00 seconds (refresh in 0.50s.)
/usr/local/lib/python3.7/dist-packages/numpy/core/fromnumeric.py:87: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
  return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
[2021/08/16 22:56:57.350][ERROR][averaging.averager._run_allreduce:426] 
Traceback (most recent call last):
  File "/content/hivemind/hivemind/averaging/averager.py", line 416, in _run_allreduce
    averaging_outputs = [output async for output in allreduce]
  File "/content/hivemind/hivemind/averaging/averager.py", line 416, in <listcomp>
    averaging_outputs = [output async for output in allreduce]
  File "/content/hivemind/hivemind/averaging/allreduce.py", line 132, in run
    async for averaged_tensor_delta in self.tensor_part_container.iterate_output_tensors():
  File "/content/hivemind/hivemind/averaging/partition.py", line 134, in iterate_output_tensors
    await self._output_part_available[peer_index].wait()
  File "/usr/lib/python3.7/asyncio/locks.py", line 293, in wait
    await fut
concurrent.futures._base.CancelledError
[2021/08/16 22:56:57.352][INFO][optim.collaborative.step:250] Skipped averaging: averaging round failed with TimeoutError().
[2021/08/16 22:56:57.368][INFO][optim.collaborative.step:266] Optimizer step: done!
```

Error log from regular peer:
```
[2021/08/16 22:55:12.202][INFO][optim.collaborative.step:229] Beginning global optimizer step 0
[2021/08/16 22:55:12.212][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 3856 samples from 2 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/16 22:56:57.266][ERROR][averaging.averager._run_allreduce:426]
Traceback (most recent call last):
  File "/storage/hdd1/jheuristic/exp/testTPU/hivemind/hivemind/averaging/averager.py", line 416, in _run_allreduce
    averaging_outputs = [output async for output in allreduce]
  File "/storage/hdd1/jheuristic/exp/testTPU/hivemind/hivemind/averaging/averager.py", line 416, in <listcomp>
    averaging_outputs = [output async for output in allreduce]
  File "/storage/hdd1/jheuristic/exp/testTPU/hivemind/hivemind/averaging/allreduce.py", line 132, in run
    async for averaged_tensor_delta in self.tensor_part_container.iterate_output_tensors():
  File "/storage/hdd1/jheuristic/exp/testTPU/hivemind/hivemind/averaging/partition.py", line 134, in iterate_output_tensors
    await self._output_part_available[peer_index].wait()
  File "/home/jheuristic/anaconda3/envs/TPU/lib/python3.9/asyncio/locks.py", line 226, in wait
    await fut
asyncio.exceptions.CancelledError
[2021/08/16 22:56:57.268][ERROR][averaging.averager._step:365] Averager caught MatchmakingException('Unable to run All-Reduce: ')
Traceback (most recent call last):
  File "/storage/hdd1/jheuristic/exp/testTPU/hivemind/hivemind/averaging/averager.py", line 416, in _run_allreduce
    averaging_outputs = [output async for output in allreduce]
  File "/storage/hdd1/jheuristic/exp/testTPU/hivemind/hivemind/averaging/averager.py", line 416, in <listcomp>
    averaging_outputs = [output async for output in allreduce]
  File "/storage/hdd1/jheuristic/exp/testTPU/hivemind/hivemind/averaging/allreduce.py", line 132, in run
    async for averaged_tensor_delta in self.tensor_part_container.iterate_output_tensors():
  File "/storage/hdd1/jheuristic/exp/testTPU/hivemind/hivemind/averaging/partition.py", line 134, in iterate_output_tensors
    await self._output_part_available[peer_index].wait()
  File "/home/jheuristic/anaconda3/envs/TPU/lib/python3.9/asyncio/locks.py", line 226, in wait
    await fut
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/storage/hdd1/jheuristic/exp/testTPU/hivemind/hivemind/averaging/averager.py", line 348, in _step
    await asyncio.wait_for(
  File "/home/jheuristic/anaconda3/envs/TPU/lib/python3.9/asyncio/tasks.py", line 492, in wait_for
    fut.result()
  File "/storage/hdd1/jheuristic/exp/testTPU/hivemind/hivemind/averaging/averager.py", line 427, in _run_allreduce
    raise MatchmakingException(f"Unable to run All-Reduce: {e}")
hivemind.averaging.matchmaking.MatchmakingException: Unable to run All-Reduce:
[2021/08/16 22:56:57.270][INFO][optim.collaborative.step:250] Skipped averaging: averaging round failed with MatchmakingException('Unable to run All-Reduce: ').
[2021/08/16 22:56:57.301][INFO][optim.collaborative.step:266] Optimizer step: done!
```

**Comment by borzunov** [comment]
**Observation 1:** Two trainers (one without and one with client mode) on nora work well.

**Observation 2:** With `LOGLEVEL=DEBUG`, a nora peer (without client mode) and a Colab peer (with client mode) fail to average with these messages (they are produced quickly):

```
[2021/08/18 23:47:07.773][DEBUG][averaging.key_manager.get_averagers:94] Allreduce group not found: my-albert-v1_averaging.0b, creating new group.
[2021/08/18 23:47:07.951][DEBUG][averaging.key_manager.get_averagers:94] Allreduce group not found: my-albert-v1_averaging.0b, creating new group.
[2021/08/18 23:47:08.129][DEBUG][averaging.key_manager.get_averagers:94] Allreduce group not found: my-albert-v1_averaging.0b, creating new group.
[2021/08/18 23:47:08.306][DEBUG][averaging.key_manager.get_averagers:94] Allreduce group not found: my-albert-v1_averaging.0b, creating new group.
[2021/08/18 23:47:08.484][DEBUG][averaging.key_manager.get_averagers:94] Allreduce group not found: my-albert-v1_averaging.0b, creating new group.
[2021/08/18 23:47:08.661][DEBUG][averaging.key_manager.get_averagers:94] Allreduce group not found: my-albert-v1_averaging.0b, creating new group.
[2021/08/18 23:47:08.838][DEBUG][averaging.key_manager.get_averagers:94] Allreduce group not found: my-albert-v1_averaging.0b, creating new group.
[2021/08/18 23:47:09.016][DEBUG][averaging.key_manager.get_averagers:94] Allreduce group not found: my-albert-v1_averaging.0b, creating new group.
```

**Comment by justheuristic** [comment]
A minor error
![image](https://user-images.githubusercontent.com/3491902/130039386-83fdea6f-8da8-4e65-a5e1-d855acc67750.png)

**Comment by justheuristic** [comment]
This task should be cancelled once the group is found - but for some reason it doesn't seem to be cancelled.
![image](https://user-images.githubusercontent.com/3491902/130046768-7a162bce-2aa5-46cd-95cc-d48b3d4717b8.png)

This supports your observation 2 above, which may imply that matchmaking does not terminate when it needs to (once _run_allreduce has begun).

**Comment by borzunov** [comment]
We have found that averaging actually works but is very slow. The problem disappears in case of disabling QUIC for both incoming and outgoing connections, as done in #355.

**Comment by borzunov** [comment]
Let's leave this open until we find a way to speed up QUIC.

**Comment by justheuristic** [comment]
Found a similar timeout pattern __occasionally__ happening in our internal infrastructure
<details>
 <summary>traceback</summary>

```
[2021/08/29 13:12:50.889][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 33576 samples from 40 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/29 13:12:51.403][INFO][optim.collaborative.step:229] Beginning global optimizer step 7
[2021/08/29 13:12:51.761][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 34496 samples from 40 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/29 13:12:51.792][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 34496 samples from 40 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/29 13:15:01.818][ERROR][averaging.averager._run_allreduce:440] 
Traceback (most recent call last):
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/averager.py", line 430, in _run_allreduce
    averaging_outputs = [output async for output in allreduce]
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/averager.py", line 430, in <listcomp>
    averaging_outputs = [output async for output in allreduce]
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/allreduce.py", line 132, in run
    async for averaged_tensor_delta in self.tensor_part_container.iterate_output_tensors():
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/partition.py", line 134, in iterate_output_tensors
    await self._output_part_available[peer_index].wait()
  File "/anaconda3/lib/python3.8/asyncio/locks.py", line 309, in wait
    await fut
asyncio.exceptions.CancelledError
[2021/08/29 13:15:01.822][INFO][optim.collaborative.step:250] Skipped averaging: averaging round failed with TimeoutError().
[2021/08/29 13:15:13.998][INFO][optim.collaborative.step:266] Optimizer step: done!
[2021/08/29 13:15:14.007][INFO][root.on_step_end:60] Step 8
[2021/08/29 13:15:14.007][INFO][root.on_step_end:61] Your current contribution: 15744 samples
[2021/08/29 13:15:14.007][INFO][root.on_step_end:62] Performance: 53.16025065774357 samples per second.
[2021/08/29 13:15:14.007][INFO][root.on_step_end:64] Local loss: 11.19378
[2021/08/29 13:15:14.340][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 208 samples from 6 peers; ETA 597.24 seconds (refresh in 30.00s.)
[2021/08/29 13:15:44.506][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 44480 samples from 42 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/29 13:15:45.181][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 45504 samples from 42 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/29 13:15:45.554][INFO][optim.collaborative.step:229] Beginning global optimizer step 8
[2021/08/29 13:15:46.044][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 45888 samples from 42 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/29 13:15:46.045][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 45888 samples from 42 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/29 13:15:48.063][ERROR][asyncio._run:259] Task exception was never retrieved
future: <Task finished name='Task-15570' coro=<DecentralizedAverager._step() done, defined at /slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/averager.py:345> exception=TimeoutError()>
Traceback (most recent call last):
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/averager.py", line 362, in _step
    await asyncio.wait_for(
  File "/anaconda3/lib/python3.8/asyncio/tasks.py", line 501, in wait_for
    raise exceptions.TimeoutError()
asyncio.exceptions.TimeoutError
[2021/08/29 13:15:53.996][INFO][__main__.analyze_openfiles_periodically:42] Scanning open files for process 3
[2021/08/29 13:15:54.004][INFO][__main__.analyze_openfiles_periodically:49] proc: 'python3' pid=3 parent=2 files: 7
[2021/08/29 13:15:54.004][INFO][__main__.analyze_openfiles_periodically:49] proc: 'python3' pid=382 parent=3 files: 2
[2021/08/29 13:15:54.007][INFO][__main__.analyze_openfiles_periodically:49] proc: 'python3' pid=562 parent=3 files: 2
[2021/08/29 13:15:54.010][INFO][__main__.analyze_openfiles_periodically:49] proc: 'python3' pid=629 parent=3 files: 7
[2021/08/29 13:15:54.013][INFO][__main__.analyze_openfiles_periodically:49] proc: 'python3' pid=757 parent=3 files: 7
[2021/08/29 13:15:54.015][INFO][__main__.analyze_openfiles_periodically:49] proc: 'python3' pid=885 parent=3 files: 7
[2021/08/29 13:15:54.016][INFO][__main__.analyze_openfiles_periodically:49] proc: 'python3' pid=1013 parent=3 files: 7
[2021/08/29 13:15:54.017][INFO][__main__.analyze_openfiles_periodically:49] proc: 'p2pd' pid=383 parent=382 files: 0
[2021/08/29 13:15:54.027][INFO][__main__.analyze_openfiles_periodically:57] DONE scanning
[2021/08/29 13:16:01.590][ERROR][asyncio._accumulate_parts_streaming:227] Future exception was never retrieved
future: <Future finished exception=CancelledError()>
Traceback (most recent call last):
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/averager.py", line 430, in _run_allreduce
    averaging_outputs = [output async for output in allreduce]
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/averager.py", line 430, in <listcomp>
    averaging_outputs = [output async for output in allreduce]
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/allreduce.py", line 132, in run
    async for averaged_tensor_delta in self.tensor_part_container.iterate_output_tensors():
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/partition.py", line 134, in iterate_output_tensors
    await self._output_part_available[peer_index].wait()
  File "/anaconda3/lib/python3.8/asyncio/locks.py", line 309, in wait
    await fut
asyncio.exceptions.CancelledError
[2021/08/29 13:16:01.592][ERROR][asyncio._accumulate_parts_streaming:227] Task exception was never retrieved
future: <Task finished name='Task-15594' coro=<DecentralizedAverager._run_allreduce() done, defined at /slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/averager.py:397> exception=MatchmakingException('Unable to run All-Reduce: ')>
Traceback (most recent call last):
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/averager.py", line 430, in _run_allreduce
    averaging_outputs = [output async for output in allreduce]
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/averager.py", line 430, in <listcomp>
    averaging_outputs = [output async for output in allreduce]
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/allreduce.py", line 132, in run
    async for averaged_tensor_delta in self.tensor_part_container.iterate_output_tensors():
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/partition.py", line 134, in iterate_output_tensors
    await self._output_part_available[peer_index].wait()
  File "/anaconda3/lib/python3.8/asyncio/locks.py", line 309, in wait
    await fut
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/averager.py", line 441, in _run_allreduce
    raise MatchmakingException(f"Unable to run All-Reduce: {e}")
hivemind.averaging.matchmaking.MatchmakingException: Unable to run All-Reduce: 
[2021/08/29 13:17:57.499][ERROR][averaging.averager._run_allreduce:440] 
Traceback (most recent call last):
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/averager.py", line 430, in _run_allreduce
    averaging_outputs = [output async for output in allreduce]
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/averager.py", line 430, in <listcomp>
    averaging_outputs = [output async for output in allreduce]
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/allreduce.py", line 132, in run
    async for averaged_tensor_delta in self.tensor_part_container.iterate_output_tensors():
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/partition.py", line 134, in iterate_output_tensors
    await self._output_part_available[peer_index].wait()
  File "/anaconda3/lib/python3.8/asyncio/locks.py", line 309, in wait
    await fut
asyncio.exceptions.CancelledError
[2021/08/29 13:17:57.503][INFO][optim.collaborative.step:250] Skipped averaging: averaging round failed with TimeoutError().
[2021/08/29 13:18:10.329][INFO][optim.collaborative.step:266] Optimizer step: done!
[2021/08/29 13:18:10.337][INFO][root.on_step_end:60] Step 9
[2021/08/29 13:18:10.338][INFO][root.on_step_end:61] Your current contribution: 17280 samples
[2021/08/29 13:18:10.338][INFO][root.on_step_end:62] Performance: 53.38832981682287 samples per second.
[2021/08/29 13:18:10.338][INFO][root.on_step_end:64] Local loss: 11.187492307692308
[2021/08/29 13:18:10.668][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 248 samples from 6 peers; ETA 565.88 seconds (refresh in 30.00s.)
[2021/08/29 13:18:40.842][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 46048 samples from 42 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/29 13:18:41.646][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 46304 samples from 43 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/29 13:18:41.917][INFO][optim.collaborative.step:229] Beginning global optimizer step 9
[2021/08/29 13:18:42.283][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 47072 samples from 43 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/29 13:18:42.295][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 47072 samples from 43 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/29 13:18:44.362][ERROR][asyncio._run:259] Task exception was never retrieved
future: <Task finished name='Task-18149' coro=<DecentralizedAverager._step() done, defined at /slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/averager.py:345> exception=TimeoutError()>
Traceback (most recent call last):
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/averaging/averager.py", line 362, in _step
    await asyncio.wait_for(
  File "/anaconda3/lib/python3.8/asyncio/tasks.py", line 501, in wait_for
    raise exceptions.TimeoutError()
asyncio.exceptions.TimeoutError
[2021/08/29 13:19:05.954][INFO][optim.collaborative.step:248] Averaged tensors successfully with 43 peers
[2021/08/29 13:19:18.521][INFO][optim.collaborative.step:266] Optimizer step: done!
[2021/08/29 13:19:18.530][INFO][root.on_step_end:60] Step 10
[2021/08/29 13:19:18.530][INFO][root.on_step_end:61] Your current contribution: 18816 samples
[2021/08/29 13:19:18.530][INFO][root.on_step_end:62] Performance: 53.40463156695707 samples per second.
[2021/08/29 13:19:18.530][INFO][root.on_step_end:64] Local loss: 11.189092307692308
[2021/08/29 13:19:18.873][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 248 samples from 43 peers; ETA 17.61 seconds (refresh in 14.68s.)
[2021/08/29 13:19:33.840][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 20784 samples from 43 peers; ETA 5.04 seconds (refresh in 4.20s.)
[2021/08/29 13:19:38.221][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 27752 samples from 43 peers; ETA 1.52 seconds (refresh in 1.27s.)
[2021/08/29 13:19:39.674][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 29816 samples from 43 peers; ETA 0.30 seconds (refresh in 0.50s.)
[2021/08/29 13:19:40.367][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 30464 samples from 43 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/29 13:19:40.595][INFO][optim.collaborative.step:229] Beginning global optimizer step 10
[2021/08/29 13:19:41.060][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 31248 samples from 43 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/29 13:19:41.062][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 31248 samples from 43 peers; ETA 0.00 seconds (refresh in 0.50s.)
[2021/08/29 13:20:09.587][INFO][optim.collaborative.step:248] Averaged tensors successfully with 42 peers
[2021/08/29 13:20:22.409][INFO][optim.collaborative.step:266] Optimizer step: done!
[2021/08/29 13:20:22.417][INFO][root.on_step_end:60] Step 11
[2021/08/29 13:20:22.417][INFO][root.on_step_end:61] Your current contribution: 19840 samples
[2021/08/29 13:20:22.417][INFO][root.on_step_end:62] Performance: 53.08986074631606 samples per second.
[2021/08/29 13:20:22.417][INFO][root.on_step_end:64] Local loss: 11.194788888888889
[2021/08/29 13:20:22.810][INFO][optim.collaborative.fetch_collaboration_state:442] Collaboration accumulated 328 sa
```

</details>

**Comment by justheuristic** [comment]
We're investigating this and brainstorming with @deniskamazur ; we've got the first positive steps!

__Measurements:__
- current master: averaging at 5-10 mbit/s (bi-dir), catching TimeoutError
- without unary handlers & without quic ([commit#](https://github.com/learning-at-home/hivemind/tree/3a77144cd737d2bb9c446120abf6619fe251939d)) - also 5-10mbit, also timeout
- Same old [commit](https://github.com/learning-at-home/hivemind/tree/3a77144cd737d2bb9c446120abf6619fe251939d) , but with  DEFAULT_PART_SIZE_BYTES = 2 ** 16 - averaging much faster at 18-30mbit and __finishing in time__

**Comment by borzunov** [comment]
An older p2pd version and `DEFAULT_PART_SIZE_BYTES = 2 ** 16` resolve the issue. They are merged in #373.

I am keeping the issue open since the original reasons of the problems remain unclear.

**Comment by justheuristic** [comment]
fixed in #385
